### PR TITLE
tools/Wasm.mk: Take optimization flags from native build

### PR DESCRIPTION
**Pull request by no1wudi**
## Summary
tools/Wasm.mk: Take optimization flags from native build
## Impact
wasm build only
## Testing
local machine

### CODE REVIEW

File: tools/Wasm.mk [hunk 1]
```diff
@@ -54,10 +54,16 @@ CFLAGS_STRIP += $(ARCHCPUFLAGS) $(ARCHCFLAGS) $(ARCHINCLUDES) $(ARCHDEFINES) $(A
 WCFLAGS += $(filter-out $(CFLAGS_STRIP),$(CFLAGS))
 WCFLAGS += --sysroot=$(WSYSROOT) -nostdlib -D__NuttX__
 
+# Keep optimization and lto flags
+
+WCFLAGS += $(filter -O%, $(ARCHOPTIMIZATION))
```
Comment (xiaoxiang781216) [inline tools/Wasm.mk:59]: should we decouple native compiler option from wasm compiler?

Comment (no1wudi) [reply tools/Wasm.mk:59]: Like this ?
```
ifeq ($(CONFIG_DEBUG_CUSTOMOPT),y)
  WCFLAGS  += $(CONFIG_DEBUG_OPTLEVEL)
else ifeq ($(CONFIG_DEBUG_FULLOPT),y)
    WCFLAGS  += -Oz
endif
```

Comment (xiaoxiang781216) [reply tools/Wasm.mk:59]: Yes, since many native options aren't usable to wasm compiler.

File: tools/Wasm.mk [hunk 2]
```diff
@@ -54,10 +54,29 @@ CFLAGS_STRIP += $(ARCHCPUFLAGS) $(ARCHCFLAGS) $(ARCHINCLUDES) $(ARCHDEFINES) $(A
 WCFLAGS += $(filter-out $(CFLAGS_STRIP),$(CFLAGS))
```
Comment (xiaoxiang781216) [inline tools/Wasm.mk:54]: let's don't reference CFLAGS too

File: tools/Wasm.mk [hunk 3]
```diff
@@ -54,10 +54,29 @@ CFLAGS_STRIP += $(ARCHCPUFLAGS) $(ARCHCFLAGS) $(ARCHINCLUDES) $(ARCHDEFINES) $(A
...
+
+ifneq ($(CONFIG_LTO_NONE),y)
+  WLDFLAGS += -Wl,--lto
+endif
+
 WLDFLAGS = -z stack-size=$(STACKSIZE) -Wl,--initial-memory=$(INITIAL_MEMORY)
 WLDFLAGS += -Wl,--export=main -Wl,--export=__main_argc_argv
 WLDFLAGS += -Wl,--export=__heap_base -Wl,--export=__data_end
 WLDFLAGS += -Wl,--no-entry -Wl,--strip-all -Wl,--allow-undefined
+WLDFLAGS += $(filter -flto%, $(ARCHOPTIMIZATION))
```
Comment (xiaoxiang781216) [inline tools/Wasm.mk:79]: remove the reference to ARCHOPTIMIZATION

**Pull request closed without merging**